cookbooks/aws-parallelcluster-config/recipes/nvidia.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+#
+# Cookbook:: aws-parallelcluster
+# Recipe:: nvidia
+#
+# Copyright:: 2013-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+if graphic_instance? && nvidia_installed?
+  # Load kernel module Nvidia-uvm
+  kernel_module 'nvidia-uvm' do
```

Comment:
why this isn't done at AMI build time, i.e. installation recipes?

Reply:
Because I want to apply these only on GPU instances.
If the logic is in AMI build: If a non-GPU instance is used, the nvidia_uvm fails to load and nvidia-persistenced service fails to start, which is expected. There are relevant error messages in /var/log/messages, but instance will launch successfully.